package/base-files/files/lib/functions/system.sh

```diff
@@ -15,7 +15,11 @@ get_mac_binary() {
 get_mac_label() {
 	local basepath="/proc/device-tree"
 	local macdevice="$(cat "$basepath/aliases/label-mac-device" 2>/dev/null)"
-	[ -n "$macdevice" ] && get_mac_binary "$basepath/$macdevice/mac-address" 0 2>/dev/null
+	local macaddr
+
+	[ -n "$macdevice" ] && macaddr=$(get_mac_binary "$basepath/$macdevice/mac-address" 0 2>/dev/null)
+	[ -n "$macaddr" ] || macaddr=$(uci -q get system.@system[-1].label_macaddr)
```

Comment:
It should be `system.@system[0].label_macaddr`

Reply:
I aligned this with how the hostname is treated in https://github.com/openwrt/openwrt/blob/master/package/base-files/files/bin/config_generate#L260. However, it won't matter in the end.
Shall I still change it?

Reply:
> However, it won't matter in the end.

It matters, you want to store it know location (section index 0) and then read it from there. With -1 index you would simply read/write it from the last section index, so it might produce error in some cases:

```
root@OpenWrt:/# uci get system.@system[0].hostname
OpenWrt

root@OpenWrt:/# uci add system system
cfg05e48a

root@OpenWrt:/# uci set system.@system[-1].hostname='LEDE'

root@OpenWrt:/# uci get system.@system[0].hostname
OpenWrt

root@OpenWrt:/# uci get system.@system[-1].hostname
LEDE
```

Reply:
I'm aware of that mechanism, I however just assumed that nobody will add a system node afterwards. I admit that _if_ someone adds such a node later, [0] will still work and [-1] won't. So I'll change to [0] ...